modules/rosjenkins/manifests/groovy.pp

```diff
@@ -0,0 +1,20 @@
+# rosjenkins::groovy
+#
+# A resource for running Groovy scripts via Jenkins CLI.
+#
+#
+define rosjenkins::groovy(
+  String $groovyfile = $title,
+  String $jarfile = '/usr/share/jenkins/jenkins-cli.jar',
+) {
+  $jenkinsauth = "${::jenkins::slave::ui_user}:${::jenkins::slave::ui_pass}"
+
+  exec { $title:
```

Comment:
Is there a test for whether this needs to be applied? An `unless` or an `onlyif` to avoid running this ever time?

Reply:
The only tests I can devise would need to also be done via groovy script.
In order to limit the amount of puppet-fu required to create change machinery, I think it's probably best to keep checks and changes in one Groovy file rather than try and separate them into unless/onlyif and mutation scripts.

This goes double since I think we would need to write an extra wrapper script to handle return values from parsing Groovy output.